tuplex/test/wrappers/WrapperTest.cc

```diff
@@ -87,6 +118,45 @@ TEST_F(WrapperTest, BasicMergeInOrder) {
     }
 }
 
+TEST_F(WrapperTest, MathIsInf) {
+    using namespace tuplex;
+
+    // create Python context (pass options as JSON)
+    PythonContext c("c", "", microTestOptions().asJSON());
+
+    // list object contains all rows in test (in this test, only one row)
+    PyObject *listObj = PyList_New(1);
+    
+    // initialize listObj
+    // note that using runAndGet on each individual value, and then setting
+    // them as an element of the list is buggy (doesn't always return the right value)
+    listObj = python::runAndGet(
+        "import math; x = [0, -1, 5, math.inf * 0, math.inf, 97]",
+        "x");
+
+    Py_XINCREF(listObj);
+    PyObject_Print(listObj, stdout, 0);
+    std::cout << std::endl;
+
+    {
+        auto list = py::reinterpret_borrow<py::list>(listObj);
+
+        auto ba_closure = PyDict_New();
+        auto math_mod = PyImport_ImportModule("math");
+        assert(math_mod);
+        PyDict_SetItemString(ba_closure, "math", math_mod);
+
+        // write parallelize function
+        auto res = c.parallelize(list).map("lambda x: math.isinf(x)", "", py::reinterpret_steal<py::dict>(ba_closure)).collect();
```

Comment:
potentially replacing reinterpret_steal with reinterpret_borrow could solve the issue